src/bundle/Templating/Twig/RecommendationExtension.php

```diff
@@ -0,0 +1,173 @@
+     *
+     * @param array $value
+     */
+    public function setIncludedContentTypes($value)
+    {
+        $this->options['includedContentTypes'] = $value;
+    }
+
+    /**
+     * Sets `customerId` option when service is created which allows to
+     * inject parameter value according to siteaccess configuration.
+     *
+     * @param string $value
+     */
+    public function setCustomerId($value)
+    {
+        $this->options['customerId'] = $value;
+    }
+
+    /**
+     * Returns the name of the extension.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return 'ez_recommendation_extension';
+    }
+
+    /**
+     * Returns a list of functions to add to the existing list.
+     *
+     * @return array
+     */
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('reco_track_user', [$this, 'trackUser'], [
+                'is_safe' => ['html'],
+                'needs_environment' => true,
+            ]),
+        ];
+    }
+
+    /**
+     * Renders simple tracking snippet code.
+     *
+     * @param \Twig\Environment $twigEnvironment
+     * @param int $contentId
+     *
+     * @return string
+     *
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function trackUser(Environment $twigEnvironment, int $contentId): string
+    {
+        if (!in_array($this->contentHelper->getContentIdentifier($contentId), $this->options['includedContentTypes'])) {
+            return '';
+        }
+
+        return $twigEnvironment->render(
+            'EzRecommendationClientBundle::track_user.html.twig',
+            [
+                'contentId' => $contentId,
+                'contentTypeId' => $this->contentHelper->getContentTypeId($this->contentHelper->getContentIdentifier($contentId)),
+                'language' => $this->localeConverter->convertToEz($this->requestStack->getCurrentRequest()->get('_locale')),
+                'userId' => $this->getUserIdentifier(),
+                'customerId' => $this->options['customerId'],
+                'consumeTimeout' => ($this->options['consumeTimeout'] * 1000),
+                'trackingScriptUrl' => $this->options['trackingScriptUrl'],
+            ]
+        );
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    protected function parseOptions(array $options): array
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        if (!empty($request) && $request->isSecure()) {
+            $options['recommenderEndPoint'] = str_replace('http://', 'https://', $options['recommenderEndPoint']);
+        }
+
+        return $options;
+    }
+
+    /** @return mixed|object|string */
```

Comment:
Is this true?

Reply:
Yes, this is true. Just look on the method getUser() from TokenInterface: Symfony\Component\Security\Core\Authentication\Token\TokenInterface

"@return string|object Can be a UserInterface instance, an object implementing a __toString method,  or the username as a regular string"